tests/contrib/test_caching.py

```diff
@@ -0,0 +1,50 @@
+import pytest
+from django.core.cache import cache
+from django.urls import reverse
+
+from adhocracy4.projects.enums import Access
+from adhocracy4.test.factories import ProjectFactory
+from meinberlin.apps.contrib import caching
+from meinberlin.test.factories.extprojects import ExternalProjectFactory
+from meinberlin.test.factories.plans import PlanFactory
+from meinberlin.test.factories.projectcontainers import ProjectContainerFactory
+
+
+def test_cache_delete():
+    ...
+
+
+def test_cache_add_or_query():
+    ...
+
+
+def test_cache_add_or_serialize():
+    ...
+
+
+@pytest.mark.django_db
+@pytest.mark.parametrize(
+    "url_name,factory,factory_kwargs",
+    [
+        ("projects-list", ProjectFactory, {}),
+        ("plans-list", PlanFactory, {}),
+        ("extprojects-list", ExternalProjectFactory, {"access": Access.PUBLIC}),
+        ("containers-list", ProjectContainerFactory, {}),
+        # ("projects-list", ProjectFactory, {"access": Access.PRIVATE}),
+    ],
+)
+def test_list_url(client, url_name, factory, factory_kwargs):
+    n_objects = 3
+    url = reverse(url_name)
+    cache_key = caching.get_key(namespace=url)
+    cache_value_before = cache.get(cache_key)
+
+    objects = factory.create_batch(size=n_objects, **factory_kwargs)
+    response = client.get(url)
+    cache_value_after = cache.get(cache_key)
+
+    assert response.status_code == 200
+    assert cache_value_before is None
+    assert len(cache_value_after) == len(objects) == n_objects
```

Comment:
check also the string pattern of the value

Reply:
that is quite tricky because the serialized object and the original model can be (and are) very different. I would have to pass each serializer to the test data. But the serializers are called differently because the `__init__(..)` has been overridden (e.g. `ProjectsSerializer` has to be called with a `now: datetime` argument, but `PlanSerializer` doesn't.

Even testing a single attribute is not straightforward as the serializer renames practically everything.

So I would suggest to stick with testing the length. Since the DB is empty at test start, it would be surprising if the cache returns the exact same number of objects, but they are different object. Right?

Reply:
I mean check that the namespace is part of the key.
But the tests are failing atm @hklarner

Reply:
The cache key is created with
```python
cache_key = caching.get_key(namespace=namespace)
```
i.e. the namespace is an argument that is required to create a key. All API view sets that have caching enabled (`ProjectViewSet`, `PlanViewSet`, etc.) have the name space hard-coded.

The test already makes the assumption that the value is saved in the correct key because it calls
```
cache_value_after = cache.get(cache_key)
```
So it already asserts that each view has the correct hard coded namespace.

Is that what you mean?

Reply:
not sure what to assert here, the code never tells me what key it uses, instead I request the cache value for a key. So the key is part of the input for the test. If I misunderstand then please add explicitly what I should assert and I will. Marking as "done" for now.